internal/git/git.go

```diff
@@ -22,36 +24,48 @@ func CloneCommandFor(s rukpakv1alpha1.GitSource) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	return cmd.String(), nil
+	return cmd.String()
 }
 
-func (c *checkoutCmd) String() string {
+func (c *checkoutCmd) String() (string, error) {
 	var checkoutCommand string
 	var repository = c.Repository
 	var directory = c.Directory
 	var branch = c.Ref.Branch
 	var commit = c.Ref.Commit
 	var tag = c.Ref.Tag
+	var preset = ""
+
+	if c.Secret != "" {
+		repositoryURL, err := url.Parse(repository)
+		if err == nil {
+			repositoryURL.User = url.UserPassword("$USER", "$TOKEN")
+			repository = repositoryURL.String()
+		}
+	}
+	if c.SslNoVerify {
+		preset = sslNoVerify
```

Comment:
nit: if this variable is being assigned only here once, maybe we can remove it entirely and inject `sslNoVerify` into the checkoutCommand directly?

Reply:
Does the above comment make sense?  Thanks!

Reply:
I see, that makes sense, you only want to conditionally set it 👍